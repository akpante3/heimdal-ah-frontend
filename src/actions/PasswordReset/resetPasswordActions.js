import Axios from 'axios';
import {
  ACTIONS,
} from '../actionTypes';
import ActionResponse from '../actionResponse';

const resetPassword = userEmail => dispatch => Axios.post(`${process.env.BASE_URL_PROD}/api/v1/password/forgot`, {
  email: userEmail,
  headers: {
    'Access-Control-Allow-Origin': '*',
  },
}).then((response) => {
  if (response.status === 200) {
    dispatch(ActionResponse(ACTIONS.RESET_PASSWORD_SUCCESS, 'Email was sent successfully'));
  } else if (response.status === 404) {
    dispatch(ActionResponse(ACTIONS.RESET_PASSWORD_FAILED, 'user not avalaible'));
  }
}).catch((error) => {
  dispatch(ActionResponse(ACTIONS.RESET_PASSWORD_ERROR, 'Server error', error));
});

export default resetPassword;
