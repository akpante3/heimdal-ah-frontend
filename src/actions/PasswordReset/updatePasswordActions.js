import Axios from 'axios';
import {
  ACTIONS,
} from '../actionTypes';
import ActionResponse from '../actionResponse';

const updatePassword = (token, newPassword) => dispatch => Axios.put(`${process.env.BASE_URL_PROD}/api/v1/password/reset/${token}`, {
  password: newPassword.password,
  confirmPassword: newPassword.confirmpassword,
}).then((response) => {
  if (response.status === 200) {
    dispatch(ActionResponse(ACTIONS.UPDATE_PASSWORD_SUCCESS, 'password update was successful'));
  } else if (response.status === 404) {
    dispatch(ActionResponse(ACTIONS.UPDATE_PASSWORD_FAILED, 'user not avalaible'));
  }
}).catch((error) => {
  dispatch(ActionResponse(ACTIONS.UPDATE_PASSWORD_ERROR, 'Server error', error));
});

export default updatePassword;
