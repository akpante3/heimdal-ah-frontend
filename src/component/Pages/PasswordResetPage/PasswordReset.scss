$default-color: #00b4fa;
$white-color: #ffffff;
$btn-google: #f13921;
$btn-facebook: #2a59a1;
$btn-most: #606e86;
$shadow: rgba(77, 56, 56, 0.3);
$shadow2: rgba(107, 105, 105, 0.2);
$shadow3: rgba(107, 105, 105, 0.8);
$reem-kufi: 'Reem Kufi', sans-serif;
$cardo: 'Cardo', serif;
$ovo: 'Ovo', serif;

.password-reset {
    * {
        box-sizing: border-box;
    }

    &.body {
        color: $btn-most;
        height: 5em;
    }

    &.button {
        font-size: 1em;
        margin: 0.3% 0 0 58.1%;
    }

    &.footer {
        background: $btn-most;
        color: $white-color;
        height: 5em;
        margin-top: 50%;
        padding: 2em;
        width: 100%;
    }

    &.header {
        background: $btn-most;
        color: $white-color;
        height: 5em;
        padding: 2em;
        width: 100%;
    }

    &.input-reset-password {
        color: $btn-most;
        font-family: $cardo;
        font-weight: bold;
        padding: 0.5em;
        width: 70%;
    }
}