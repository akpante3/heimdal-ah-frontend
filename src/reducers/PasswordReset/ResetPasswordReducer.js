import {
  ACTIONS,
} from '../../actions/actionTypes';

const initialState = {
  payload: '',
  status: '',
};

export default (state = initialState, action) => {
  switch (action.type) {
    case ACTIONS.START_LOADING:
      return ({
        ...state,
        payload: action.payload,
        status: 'LOADING',
      });

    case ACTIONS.STOP_LOADING:
      return ({
        ...state,
        payload: action.payload,
        status: 'LOADING_FINISHED',
      });

    case ACTIONS.RESET_PASSWORD_FAILED:
      return ({
        ...state,
        payload: action.payload,
        status: 'FAILED',
      });

    case ACTIONS.RESET_PASSWORD_SUCCESS:
      return ({
        ...state,
        payload: action.payload,
        status: 'SUCCESS',
      });

    case ACTIONS.RESET_PASSWORD_ERROR:
      return ({
        ...state,
        payload: action.payload,
        status: 'ERROR',
      });

    default:
      return (state);
  }
};
