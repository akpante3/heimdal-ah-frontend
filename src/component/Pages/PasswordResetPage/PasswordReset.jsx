import { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import './PasswordReset.scss';
import resetPassword from '../../../actions/PasswordReset/resetPasswordActions';

/**
 * @description - Helps a user resets his password
 * @param {props} status - the status returned from dispatching actions
 * @param {props} resetPassword - the action been dispatched
 * @returns {component} Component
 */
export class PasswordReset extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
    };
    this.onHandleChange = this.onHandleChange.bind(this);
    this.onHandleSubmit = this.onHandleSubmit.bind(this);
  }

  /**
 * @description - Handles the form submit by calling resetPassword action
 * @param {event} e - the event passed to the method
 * @returns {component} null
 */
  onHandleChange(e) {
    this.setState({
      [e.target.name]: e.target.value,
    });
  }

  /**
 * @description - Handles the form submit by calling resetPassword action
 * @param {event} e - the event passed to the method
 * @returns {component} null
 */
  onHandleSubmit(e) {
    const { email } = this.state;
    e.preventDefault();
    // eslint-disable-next-line react/destructuring-assignment
    this.props.actions(resetPassword(email));
  }

  /**
 * @description - Handles the rendering of the component
 * @returns {component} the rendered component
 */
  render() {
    const { email } = this.state;
    const { status } = this.props;
    return this.props.children({
      email,
      status,
      state: this.state,
      emailField: this.emailField,
      onHandleSubmit: this.onHandleSubmit,
      onHandleChange: this.onHandleChange,
    });
  }
}

PasswordReset.propTypes = {
  status: PropTypes.string.isRequired,
};

const mapStateToProps = state => ({
  status: state.resetpassword.status,
});


const mapDispatchToProps = dispatch => ({
  actions: action => dispatch(action),
});

export default connect(mapStateToProps, mapDispatchToProps)(PasswordReset);
