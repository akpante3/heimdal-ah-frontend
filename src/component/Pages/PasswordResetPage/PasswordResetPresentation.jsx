import React, { Fragment } from 'react';
import Button from '../../ui/Buttons/Button';
import Header from '../../ui/header/Header';
import Footer from '../../ui/footer/Footer';
import Alert from '../../ui/Alert/Alert';
import './PasswordReset.scss';
import PasswordReset from './PasswordReset';

const PasswordResetPresentation = () => (
  <PasswordReset>
    {
        data => (
          <Fragment>
            {data.status === 'SUCCESS' && <Alert type='success' title='Email Sent' message='An Email has been sent to you, please click on the link to reset your password' />}
            {data.status === 'FAILED' && <Alert type='warning' title='Invalid Credentials' message='User not Found' />}
            <Header isValidated={false} />
            <div className='password-reset body'>
              <p>Reset Your Password</p>
              <form onSubmit={e => data.onHandleSubmit(e)}>
                <input type='email' className='password-reset input-reset-password' name='email' value={data.email} placeholder='Enter your Email' onChange={e => data.onHandleChange(e)} required />
                <br />
                {data.status === 'ERROR' && <Alert type='warning' title='Email not sent' message='Server Error' />}
                <br />
                <Button type='login2' label='Reset' Class='password-reset button' />
              </form>
            </div>
            <br />
            <Footer />
          </Fragment>
        )
    }
  </PasswordReset>
);


export default PasswordResetPresentation;
