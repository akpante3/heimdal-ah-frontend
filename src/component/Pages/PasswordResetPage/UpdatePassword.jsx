import React, { Fragment, Component } from 'react';
import { Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import { toastr } from 'react-redux-toastr';
import PropTypes from 'prop-types';
import Header from '../../ui/header/Header';
import Footer from '../../ui/footer/Footer';
import Button from '../../ui/Buttons/Button';
import './PasswordReset.scss';
import updatePassword from '../../../actions/PasswordReset/updatePasswordActions';

/**
 * @description - Helps a user updates his password
 * @param {props} status - the status returned from dispatching actions
 * @param {props} updatePassword - the action been dispatched
 * @returns {component} Component
 */
class PasswordUpdate extends Component {
  state = {
    password: '',
    confirmpassword: '',
  }

  /**
 * @description - Handles the form input change
 * @param {event} e - the event passed to the method
 * @returns {component} updated state
 */
  onHandleChange(e) {
    this.setState({
      [e.target.name]: e.target.value,
    });
  }

  /**
 * @description - Handles the form submit by calling updatePassword action
 * @param {event} e - the event passed to the method
 * @returns {component} null
 */
  onHandleSubmit(e) {
    const { password, confirmpassword } = this.state;
    e.preventDefault();
    const token = new URLSearchParams(window.location.search).get('token');
    if (password === confirmpassword) {
      toastr.success('Loading...', 'Loading please be patient');
      // eslint-disable-next-line react/destructuring-assignment
      this.props.updatePassword(token, { password, confirmpassword });
    } else {
      toastr.warning('Password', 'Password and Confirm passsword must be the same');
    }
  }

  /**
 * @description - Handles the rendering of the component
 * @returns {component} the rendered component
 */
  render() {
    const { password, confirmpassword } = this.state;
    const { status } = this.props;
    return (
      <Fragment>
        {status === 'SUCCESS' && <Redirect to='/login' />}
        {status === 'FAILED' && <Redirect to='/resetpassword' />}
        <Header isValidated={false} />
        <div className='password-reset body'>
          <p>Enter Your New Password</p>
          {status === 'ERROR' && toastr.warning('Error', 'Server Error')}
          <form onSubmit={e => this.onHandleSubmit(e)}>
            <input type='text' className='password-reset input-reset-password' name='password' value={password} placeholder='Password' onChange={e => this.onHandleChange(e)} required />
            <br />
            <br />
            <input type='text' className='password-reset input-reset-password' name='confirmpassword' value={confirmpassword} placeholder='Confirm Password' onChange={e => this.onHandleChange(e)} required />
            <br />
            <br />
            <Button type='login2' label='Update' Class='password-reset button' />
          </form>
        </div>
        <br />
        <Footer />
      </Fragment>
    );
  }
}

PasswordUpdate.propTypes = {
  status: PropTypes.string.isRequired,
  updatePassword: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  status: state.updatepassword.status,
});

export default connect(mapStateToProps, { updatePassword })(PasswordUpdate);
