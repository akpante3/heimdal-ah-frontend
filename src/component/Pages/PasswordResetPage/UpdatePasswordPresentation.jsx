import React, { Fragment } from 'react';
import { Redirect } from 'react-router-dom';
import Button from '../../ui/Buttons/Button';
import Header from '../../ui/header/Header';
import Footer from '../../ui/footer/Footer';
import Alert from '../../ui/Alert/Alert';
import './PasswordReset.scss';
import PasswordUpdate from './UpdatePassword';

const UpdatePasswordPresentation = () => (
  <PasswordUpdate>
    {
        data => (
          <Fragment>
            {data.status === 'SUCCESS' && <Redirect to='/login' />}
            {data.status === 'FAILED' && <Redirect to='/resetpassword' />}
            <Header isValidated={false} />
            <div className='password-reset body'>
              <p>Enter Your New Password</p>
              {data.status === 'ERROR' && <Alert type='warning' title='Email not sent' message='Server Error' />}
              <form onSubmit={e => data.onHandleSubmit(e)}>
                <input type='text' className='password-reset input-reset-password' name='password' value={data.password} placeholder='Password' onChange={e => data.onHandleChange(e)} required />
                <br />
                <br />
                <input type='text' className='password-reset input-reset-password' name='confirmpassword' value={data.confirmpassword} placeholder='Confirm Password' onChange={e => data.onHandleChange(e)} required />
                <br />
                <br />
                <Button type='login2' label='Update' Class='password-reset button' />
              </form>
            </div>
            <br />
            <Footer />
          </Fragment>
        )
    }
  </PasswordUpdate>
);


export default UpdatePasswordPresentation;
