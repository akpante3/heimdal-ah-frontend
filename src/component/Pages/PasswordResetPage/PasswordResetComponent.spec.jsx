/* eslint-disable react/no-children-prop */
/* eslint-disable no-undef */
import React from 'react';
import sinon from 'sinon';
import PasswordResetPresentation from './PasswordResetPresentation';
import UpdatePasswordPresentation from './UpdatePasswordPresentation';
import { PasswordUpdate } from './UpdatePassword';
import { PasswordReset } from './PasswordReset';
import resetPassword from '../../../actions/PasswordReset/resetPasswordActions';
import updatePassword from '../../../actions/PasswordReset/updatePasswordActions';

describe('Password Reset Component Test Suite', () => {
  describe('<PasswordReset />', () => {
    it('should render the PasswordResetPresentation component', () => {
      const wrapper = shallow(<PasswordResetPresentation />);
      expect(wrapper.length).toBe(1);
      expect(wrapper).toMatchSnapshot();
    });

    it('should render the PasswordReset component', () => {
      // eslint-disable-next-line react/no-children-prop
      const wrapper = shallow(<PasswordReset status="SUCCESS" children={() => {}} />);
      expect(wrapper.length).toBe(1);
      expect(wrapper).toMatchSnapshot();
    });

    it('should handle submit events', () => {
      // eslint-disable-next-line react/no-children-prop
      const wrapper = shallow(
        <PasswordReset status="SUCCESS" children={() => {}} actions={() => {}} />,
      );
      const e = { target: { name: 'email', value: 'Heimdal' }, preventDefault: () => {} };
      wrapper.instance().onHandleSubmit(e);
      expect(
        sinon.stub(jest.fn(wrapper.instance().props.actions(resetPassword(e.target.value)))),
      ).not.toHaveBeenCalled();
    });

    it('should handle change events', () => {
      // eslint-disable-next-line react/no-children-prop
      const wrapper = shallow(<PasswordReset status="SUCCESS" children={() => {}} />);
      const e = { target: { name: 'email', value: 'Heimdal' }, preventDefault: () => {} };

      expect(wrapper.state().email).toEqual('');
      wrapper.instance().onHandleChange(e);
      expect(wrapper.instance().state.email).toEqual(e.target.value);
    });
  });

  describe('<UpdatePassword />', () => {
    it('should render the UpdatePasswordPresentation component', () => {
      const wrapper = shallow(<UpdatePasswordPresentation />);
      expect(wrapper.length).toBe(1);
      expect(wrapper).toMatchSnapshot();
    });

    it('should render the UpdatePassword component', () => {
      // eslint-disable-next-line react/no-children-prop
      const wrapper = shallow(
        <PasswordUpdate status="SUCCESS" children={() => {}} actions={() => {}} />,
      );
      expect(wrapper.length).toBe(1);
      expect(wrapper).toMatchSnapshot();
    });

    it('should handle submit events', () => {
      // eslint-disable-next-line react/no-children-prop
      const wrapper = shallow(
        <PasswordUpdate status="SUCCESS" children={() => {}} actions={() => {}} />,
      );
      const e = {
        target: { name: 'password', value: 'HeimdalPassword' },
        preventDefault: () => {},
      };
      wrapper.instance().onHandleSubmit(e);
      const newPassword = {
        password: e.target.value,
        confirmpassword: e.target.value,
      };
      const token = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c';
      // expect(jest.fn(wrapper.instance().props.actions(updatePassword(token, newPassword))))
      //   .not.toHaveBeenCalled();
      expect(
        sinon.stub(jest.fn(wrapper.instance().props.actions(updatePassword(token, newPassword)))),
      ).not.toHaveBeenCalled();
    });

    it('should handle change events', () => {
      // eslint-disable-next-line react/no-children-prop
      const wrapper = shallow(
        <PasswordUpdate status="SUCCESS" children={() => {}} actions={() => {}} />,
      );
      const e = {
        target: { name: 'password', value: 'HeimdalPassword' },
        preventDefault: () => {},
      };
      expect(wrapper.state().password).toEqual('');
      expect(wrapper.state().confirmpassword).toEqual('');
      wrapper.instance().onHandleChange(e);
      expect(wrapper.instance().state.password).toEqual(e.target.value);
    });
  });
});
