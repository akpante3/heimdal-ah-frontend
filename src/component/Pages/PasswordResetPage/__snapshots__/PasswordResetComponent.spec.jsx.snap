// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Password Reset Component Test Suite <PasswordReset /> should render the PasswordReset component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PasswordReset
    status="SUCCESS"
  >
    [Function]
  </PasswordReset>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): undefined,
  Symbol(enzyme.__nodes__): Array [
    undefined,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Password Reset Component Test Suite <PasswordReset /> should render the PasswordResetPresentation component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PasswordResetPresentation />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Password Reset Component Test Suite <UpdatePassword /> should render the UpdatePassword component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PasswordUpdate
    actions={[Function]}
    status="SUCCESS"
  >
    [Function]
  </PasswordUpdate>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): undefined,
  Symbol(enzyme.__nodes__): Array [
    undefined,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Password Reset Component Test Suite <UpdatePassword /> should render the UpdatePasswordPresentation component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UpdatePasswordPresentation />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
